SYMBOL_TABLE_ENTRY_CREATION.md

```diff
@@ -0,0 +1,354 @@
+
+**GenSymEntry**
+```chapel
+class NewSymEntry:GenSymEntry {
+    // Properties specific to the class
+    
+    proc init() {
+        super.init(elementType, objectSize);
+        this.entryType = SymbolEntryType.SegArraySymEntry;
+        assignableTypes.add(this.entryType);
+    }
+    
+    override proc getSizeEstimate(): int {
+    
+    }
+}
+```
+
+**CompositeSymEntry**
+```chapel
+class NewSymEntry:CompositeSymEntry {
+    // Properties specific to the class
+    
+    proc init() {
+        super.init(size);
+        this.entryType = SymbolEntryType.EntryType;
+        assignableTypes.add(this.entryType);
+    }
+    
+    override proc getSizeEstimate(): int {
+    
+    }
+}
+```
+
+<a id="enum"></a>
+### Add to `SymbolEntryType` Enum
+The `SymbolEntryType` enum is used to keep track of Symbol Table entry types. When a entry is initialized, the enum value is stored to the `assignableTypes` property. This property is then used to determine valid Symbol Table entry types that an entry can be cast to. When initially access a Symbol Table entry, usually an `AbstractSymEntry` is returned. This class contains the `assignableTypes` property and allows for easy identification of the entry's valid types.
+
+As an example, let's look at adding a `NewSymEntry` that inherits from `CompositeSymEntry` to the enum object. You'll notice when looking at the object that indentation is used to indicate parent-child relations. The result of adding `NewSymEntry` is below:
+
+```chapel
+enum SymbolEntryType {
+    AbstractSymEntry,  // Root Type from which all other types will inherit
+    
+        TypedArraySymEntry, // Parent type for Arrays with a dtype, legacy->GenSymEntry
+            PrimitiveTypedArraySymEntry, // int, uint8, bool, etc.
+            ComplexTypedArraySymEntry,   // DateTime, TimeDelta, IP Address, etc.
+    
+        GenSymEntry,
+            SegStringSymEntry,    // SegString composed of offset-int[], bytes->uint(8)
+            CategoricalSymEntry,  // Categorical
+            SegArraySymEntry,     // Segmented Array
+
+        CompositeSymEntry,        // Entries that consist of multiple SymEntries of varying type
+            GroupBySymEntry,      // GroupBy
+            NewSymEntry,          // Your New Symbol Table Entry
+
+        AnythingSymEntry, // Placeholder to stick aritrary things in the map
+        UnknownSymEntry,
+        None
+}
+```
+
+<a id="props"></a>
+### Adding Properties
+In addition to its inherited properties, your new Symbol Table entry will need to define properties specific to itself. In most cases, these will be `SymEntry` objects contained within this entry. 
+
+Let's look at an example where we have 2 `SymEntry` objects as properties. The first will be an array of strings and the other an array of integers. First, we need to highlight a few assumptions:
```

Comment:
First, we need to highlight a few assumptions:
->
For our example:

Reply:
done